main.go

```diff
@@ -114,20 +115,29 @@ func main() {
 			if (*flagMaxFails > 0) && (fails >= *flagMaxFails) {
 				atomic.StoreInt32(&exitFlag, 1)
 			}
-			f, err := ioutil.TempFile("", "go-stress")
-			if err != nil {
-				fmt.Printf("failed to create temp file: %v\n", err)
-				os.Exit(1)
-			}
-			f.Write(out)
-			f.Close()
-			if len(out) > 2<<10 {
-				out = out[:2<<10]
+			if *flagStdErr {
+				fmt.Fprintf(os.Stderr, "\n%s\n", out)
+			} else {
+				f, err := ioutil.TempFile("", "go-stress")
+				if err != nil {
+					fmt.Printf("failed to create temp file: %v\n", err)
+					os.Exit(1)
+				}
+				f.Write(out)
+				f.Close()
+				if len(out) > 2<<10 {
```

Comment:
Alternate suggestion: make this `2<<10` configurable as `--maxoutput`. Then you could do `--maxoutput=0` to get the filenames alone, or `--maxoutput=$LOTS` instead of `--stderr` to get everything.